File: chrono-cl - Users/Modelo/TControlHoras.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Data.SqlServerCe;


namespace CL_CHRONO.Modelo
{
    public class TControlHoras
    {
        [Key]
        public int IdControlHoras { get; set; }
        public decimal TotalHoras { get; set; }
        public DateTime Fecha { get; set; }
        public string Token { get; set; }

        // Constructor vacío
        public TControlHoras() { }

        // Constructor con parámetros
        public TControlHoras(int idControlHoras, decimal totalHoras, DateTime fecha, string token)
        {
            IdControlHoras = idControlHoras;
            TotalHoras = totalHoras;
            Fecha = fecha;
            Token = token;
        }

        // Método opcional para obtener una representación en string del objeto
        public override string ToString()
        {
            return $"Id: {IdControlHoras}, Horas: {TotalHoras}, Fecha: {Fecha.ToShortDateString()}, Token: {Token}";
        }
    }
}

File: chrononet/lib/Modelo/t_controlhoras.dart
class TControlHoras {
  final int idControlHoras;       // ID de control de horas
  final double totalHoras;        // Total de horas trabajadas (decimal -> double)
  final DateTime fecha;           // Fecha asociada
  final String token;             // Token asociado

  // Constructor
  TControlHoras({
    required this.idControlHoras,
    required this.totalHoras,
    required this.fecha,
    required this.token,
  });

  // Constructor vacío opcional
  TControlHoras.empty()
      : idControlHoras = 0,
        totalHoras = 0.0,
        fecha = DateTime.now(),
        token = '';

  // Método para convertir desde JSON a un objeto Dart
  factory TControlHoras.fromJson(Map<String, dynamic> json) {
    return TControlHoras(
      idControlHoras: json['idControlHoras'],
      totalHoras: (json['totalHoras'] as num).toDouble(),
      fecha: DateTime.parse(json['fecha']),
      token: json['token'],
    );
  }

  // Método para convertir un objeto Dart a JSON
  Map<String, dynamic> toJson() {
    return {
      'idControlHoras': idControlHoras,
      'totalHoras': totalHoras,
      'fecha': fecha.toIso8601String(),
      'token': token,
    };
  }

  // Representación en cadena (similar a ToString en C#)
  @override
  String toString() {
    return 'Id: $idControlHoras, Horas: $totalHoras, Fecha: ${fecha.toLocal()}, Token: $token';
  }
}
